Description: Only show monitor labels when display applet has focus
Author: Robert Ancell <robert.ancell@canonical.com>
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=589562
Bug-Ubuntu: https://bugs.launchpad.net/bugs/403840

Index: gnome-control-center-2.32.1/capplets/display/xrandr-capplet.c
===================================================================
--- gnome-control-center-2.32.1.orig/capplets/display/xrandr-capplet.c	2011-03-17 14:49:29.717410905 +1100
+++ gnome-control-center-2.32.1/capplets/display/xrandr-capplet.c	2011-03-17 14:49:27.167397090 +1100
@@ -149,6 +149,8 @@
     }
 
     app->labeler = gnome_rr_labeler_new (app->current_configuration);
+	if (gtk_widget_has_focus (app->dialog))
+		gnome_rr_labeler_show (app->labeler);
 
     select_current_output_from_dialog_position (app);
 }
@@ -2312,6 +2314,26 @@
     return FALSE;
 }
 
+static gboolean
+dialog_focus_in_cb (GtkWidget *widget, GdkEvent *event, gpointer data)
+{
+    App *app = data;
+
+	if (app->labeler)
+		gnome_rr_labeler_show (app->labeler);
+    return FALSE;
+}
+
+static gboolean
+dialog_focus_out_cb (GtkWidget *widget, GdkEvent *event, gpointer data)
+{
+    App *app = data;
+
+	if (app->labeler)
+		gnome_rr_labeler_hide (app->labeler);
+    return FALSE;
+}
+
 static void
 hide_help_button (App *app)
 {
@@ -2466,6 +2488,10 @@
     app->dialog = _gtk_builder_get_widget (builder, "dialog");
     g_signal_connect_after (app->dialog, "map-event",
 			    G_CALLBACK (dialog_map_event_cb), app);
+    g_signal_connect_after (app->dialog, "focus-in-event",
+			    G_CALLBACK (dialog_focus_in_cb), app);
+    g_signal_connect_after (app->dialog, "focus-out-event",
+			    G_CALLBACK (dialog_focus_out_cb), app);
 
     gtk_window_set_default_icon_name ("preferences-desktop-display");
     gtk_window_set_icon_name (GTK_WINDOW (app->dialog),
